lib/rgb_matrix/engine.ex

```diff
@@ -50,62 +44,88 @@ defmodule RGBMatrix.Engine do
   end
 
   @doc """
-  Register a `RGBMatrix.Paintable` for the engine to paint pixels to.
-  This function is idempotent.
+  Register a paint function for the engine to send frames to. This function is
+  idempotent.
   """
-  @spec register_paintable(paintable :: module) :: :ok
-  def register_paintable(paintable) do
-    GenServer.call(__MODULE__, {:register_paintable, paintable})
+  @spec register_paintable(key :: any, paint_fn :: function) :: :ok
+  def register_paintable(key, paint_fn) do
+    GenServer.call(__MODULE__, {:register_paintable, {key, paint_fn}})
   end
 
   @doc """
-  Unregister a `RGBMatrix.Paintable` so the engine no longer paints pixels to
-  it. This function is idempotent.
+  Unregister a paint function so the engine no longer sends frames to it. This
+  function is idempotent.
   """
-  @spec unregister_paintable(paintable :: module) :: :ok
-  def unregister_paintable(paintable) do
-    GenServer.call(__MODULE__, {:unregister_paintable, paintable})
+  @spec unregister_paintable(key :: any) :: :ok
+  def unregister_paintable(key) do
+    GenServer.call(__MODULE__, {:unregister_paintable, key})
   end
 
+  @doc """
+  Sends an interaction events to the engine. Animations may or may not respond
+  to these interaction events.
+  """
   @spec interact(led :: LED.t()) :: :ok
   def interact(led) do
     GenServer.cast(__MODULE__, {:interact, led})
   end
 
+  @doc """
+  Retrieves the current animation's configuration and configuration schema.
+  """
+  @spec get_animation_config() :: {config :: any, config_schema :: any}
+  def get_animation_config do
+    GenServer.call(__MODULE__, :get_animation_config)
+  end
```

Comment:
only the engine state has access to the active animation struct though...

Reply:
That's exactly my point. I'm not sure that the engine should be responsible for returning that info to other components. I think it may be an application-level concern. See the code here: https://github.com/ElixirSeattle/xebow/pull/56#discussion_r450631601